### PR TITLE
fix(#9294): adds deploy-info to service worker

### PR DESCRIPTION
**Pull request by dianabarsan**
# Description

Uses service worker deploy-info to get version in feedback service. 

#9294 
#8645

# Code review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [ ] Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# Compose URLs
If Build CI hasn't passed, these may 404:

* [Core](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:9294-feedback-doc-on-couch-crash/docker-compose/cht-core.yml)
* [CouchDB Single](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:9294-feedback-doc-on-couch-crash/docker-compose/cht-couchdb.yml)
* [CouchDB Cluster](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:9294-feedback-doc-on-couch-crash/docker-compose/cht-couchdb-clustered.yml)

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

**Comment by dianabarsan** [comment]
Since this covers some of https://github.com/medic/cht-core/issues/8645, I will redact that issue title and description to include what is being created here.

**Comment by dianabarsan** [comment]
Hi @latin-panda . Would you mind giving this a read? Thanks!

**Comment by latin-panda** [comment]
Sure @dianabarsan! I can review it tomorrow morning if that's okay, I'm packed today. In the meantime, can you also assign a QA Engineer? Maybe Lorena

File: webapp/src/ts/services/version.service.ts [hunk 1]
```diff
@@ -35,6 +38,19 @@ export class VersionService {
       });
   }
 
+  getServiceWorker ():Promise<{ version }> {
```
Comment (latin-panda) [inline webapp/src/ts/services/version.service.ts:41]:
```suggestion
  getServiceWorker():Promise<{ version }> {
```

File: webapp/src/ts/services/version.service.ts [hunk 2]
```diff
@@ -35,6 +38,19 @@ export class VersionService {
...
+    return new Promise((resolve, reject) => {
+      this.http
+        .get('/deploy-info.json')
```
Comment (latin-panda) [inline webapp/src/ts/services/version.service.ts:44]: Another alternative that seems cleaner:
```suggestion
        .get('/deploy-info.json')
        .toPromise()
        .then(....
```

**Review by latin-panda — APPROVED** [review]
Looks good to me, I left 2 tiny comments. Approving now to unblock
I did a smoke test and it works
Screenshot 2024-08-14 at 6 25 00 PM
Screenshot 2024-08-14 at 6 33 28 PM

**Review by lorerod — APPROVED** [review]
Thanks @dianabarsan this looks great. 
I also reproduced and confirm the fix for [No feedback document recorded on PouchDb crash](https://github.com/medic/cht-core/issues/9294#top).

File: webapp/src/ts/services/version.service.ts (hunk 2 above)
Comment (dianabarsan) [reply webapp/src/ts/services/version.service.ts:44]: `toPromise` is deprecated. Switched to using `firstValueFrom` instead.

**Pull request merged**